app/javascript/mastodon/features/notifications/components/setting_toggle.js

```diff
@@ -3,12 +3,28 @@ import PropTypes from 'prop-types';
 import ImmutablePropTypes from 'react-immutable-proptypes';
 import Toggle from 'react-toggle';
 
-const SettingToggle = ({ settings, settingKey, label, onChange, htmlFor = '' }) => (
-  <label htmlFor={htmlFor} className='setting-toggle__label'>
-    <Toggle checked={settings.getIn(settingKey)} onChange={(e) => onChange(settingKey, e.target.checked)} />
-    <span className='setting-toggle'>{label}</span>
-  </label>
-);
+class SettingToggle extends React.PureComponent {
```

Comment:
https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=true&presets=stage-2&targets=&browsers=last%202%20chrome%20versions&builtIns=false&debug=false&code=class%20A%20%7B%0A%20%20method%20%3D%20()%20%3D%3E%20%7B%0A%20%20%20%20console.log('a')%3B%0A%20%20%7D%0A%20%20%0A%20%20render()%20%7B%0A%20%20%20%20this.method()%3B%0A%20%20%7D%0A%7D%0A%0Aconst%20B%20%3D%20()%20%3D%3E%20%7B%0A%20%20const%20method%20%3D%20()%20%3D%3E%20%7B%0A%20%20%20%20console.log('b')%3B%0A%20%20%7D%3B%0A%0A%20%20return%20method()%3B%0A%7D%0A

As far as I can see, the two seem to be similar.